pyramid/wr_file.go

```diff
@@ -12,28 +12,56 @@ type WRFile struct {
 
 	persisted bool
 	store     func(string) error
+	abort     func() error
+	aborted   bool
 }
 
-var errAlreadyPersisted = errors.New("file is already persisted")
+var (
+	errFilePersisted = errors.New("file is persisted")
+	errFileAborted   = errors.New("file is aborted")
+)
 
 // Store copies the closed file to all tiers of the pyramid.
 func (f *WRFile) Store(filename string) error {
+	if f.aborted {
+		return errFileAborted
+	}
+	if f.persisted {
+		return errFilePersisted
+	}
+	f.persisted = true
+
 	if err := validateFilename(filename); err != nil {
 		return err
 	}
 
-	err := f.File.Close()
-	if err != nil && !errors.Is(err, os.ErrClosed) {
-		return fmt.Errorf("closing file: %w", err)
+	if err := f.idempotentClose(); err != nil {
+		return err
 	}
 
+	return f.store(filename)
+}
+
+// Abort delete the file and cleans all traces of it.
+// If file was already stored, returns an error.
+func (f *WRFile) Abort() error {
 	if f.persisted {
```

Comment:
want to check c.aborted first and return

Reply:
Why? I'm fine with letting it fail if the file isn't there. Like any other misplaced operation on an os file handler.

Reply:
it just looks easier to protect the call using the bool you already have and not having the user try to understand if a resource already aborted

Reply:
You have a bug anyhow if you call multiple aborts, so I think custom pyramid error is the same as io.ErrFileNotFound